### PR TITLE
fix(joins): Fix time query time range calculation for composite queries

### PR DESCRIPTION
**Pull request by fpacifici**
This is now looking for the longest time range in all subqueries.

It also looks for complex conditions in all subqueries instead of assuming one query only

### CODE REVIEW

File: snuba/clickhouse/query_inspector.py [hunk 1]
```diff
@@ -38,6 +88,16 @@ def _visit_join(self, data_source: JoinClause[Table]) -> None:
         self.visit_join_clause(data_source)
 
     def _visit_simple_query(self, data_source: ProcessableQuery[Table]) -> None:
+        time_range = _get_date_range(data_source)
+        if time_range and (
+            self.__max_time_range is None or time_range > self.__max_time_range
+        ):
+            self.__max_time_range = time_range
+
+        self.__has_complex_conditions = (
```
Comment (evanh) [inline snuba/clickhouse/query_inspector.py:97]: Should this only happen if `__max_time_range` is updated?

**Review by evanh — COMMENTED** [review]
No tests?

**Comment by fpacifici** [comment]
https://github.com/getsentry/snuba/pull/1630/files#diff-7b9af362f4f8a5b0309b319a1a21fb111000eb3314b63f9b3863599d862ab948R128-R141

This tests the time range condition for composite query.

File: snuba/clickhouse/query_inspector.py (hunk 1 above)
Comment (fpacifici) [reply snuba/clickhouse/query_inspector.py:97]: I don't think so. Complex condition and max time range are independent

**Review by evanh — APPROVED** [review]

**Pull request merged**